old/bin/aa.pl

```diff
@@ -156,12 +156,81 @@ sub edit {
         } else {
             $form->error("Unknown AR/AP selection value: $form->{ARAP}");
         }
+    }
+
+    &display_form;
+}
+
+sub reverse {
+    $form->{title}     = $locale->text('Add');
+    $form->{invnumber} .= '-VOID';
 
+    &create_links;
+
+    $form->{reversing} = delete $form->{id};
+    delete $form->{approved};
+    $form->{reverse} = $form->{reverse} ? 0 : 1;
+    $form->{paidaccounts} = 0;
+    if ($form->{reverse}){
+        if ($form->{ARAP} eq 'AR'){
+            $form->{subtype} = 'credit_note';
+            $form->{type} = 'transaction';
+        } elsif ($form->{ARAP} eq 'AP'){
+            $form->{subtype} = 'debit_note';
+            $form->{type} = 'transaction';
+        } else {
+            $form->error("Unknown AR/AP selection value: $form->{ARAP}");
+        }
     }
 
     &display_form;
 }
 
+sub post_reversing {
+    # we should save only the reference, sequence, transdate, description and notes;
+    # get the rest from the transaction being reversed.
+
+    $form->error(
+        $locale->text('Cannot post transaction for a closed period!') )
+        if ( $transdate and $form->is_closed( $transdate ) );
+    if (not $form->{id}) {
+        do {
+            local $form->{id} = $form->{reversing};
+
+            # save data we want to use from the posted form,
+            # not from the reversed transaction.
+            local $form->{reversing};
+            local $form->{reverse};
+            local $form->{notes};
+            local $form->{description};
+            local $form->{transdate};
+            local $form->{reference};
+            local $form->{approved};
+
+            &create_links; # create_links overwrites 'reversing'
+        };
+
+        AA->post_transaction( \%myconfig, \%$form );
```

Comment:
The unfortunate side-effect of this choice is that the transaction goes through all the logic of loading and posting a transaction instead of copying the original transaction and negating its effect. I'm looking for ideas on how to bypass the loading and saving to prevent "damage" to the exact (negated) copy.